From: Marc Kleine-Budde <mkl@pengutronix.de>
Date: Fri, 24 Apr 2009 18:44:11 +0200
Subject: [PATCH] use AC_CHECK_SIZEOF rather than AC_TRY_COMPILE for long long detection

AC_CHECK_SIZEOF does first detect if the type is available and detects
its size. Use it, rather than hand crafted function with AC_TRY_COMPILE.

Signed-off-by: Marc Kleine-Budde <mkl@pengutronix.de>
---
 configure.in |   12 +++---------
 1 files changed, 3 insertions(+), 9 deletions(-)

diff --git a/configure.in b/configure.in
index 8d130b4..c85e89b 100644
--- a/configure.in
+++ b/configure.in
@@ -1455,15 +1455,9 @@ AC_CHECK_SIZEOF(fpos_t, 4)
 AC_CHECK_SIZEOF(size_t, 4)
 AC_CHECK_SIZEOF(pid_t, 4)
 
-AC_MSG_CHECKING(for long long support)
-have_long_long=no
-AC_TRY_COMPILE([], [long long x; x = (long long)0;], [
-  AC_DEFINE(HAVE_LONG_LONG, 1, [Define this if you have the type long long.]) 
-  have_long_long=yes
-])
-AC_MSG_RESULT($have_long_long)
-if test "$have_long_long" = yes ; then
-AC_CHECK_SIZEOF(long long, 8)
+AC_CHECK_SIZEOF(long long)
+if test "$ac_cv_type_long_long" = "yes" ; then
+  AC_DEFINE(HAVE_LONG_LONG, 1, [Define this if you have the type long long.])
 fi
 
 AC_MSG_CHECKING(for long double support)
-- 
1.7.2.3

